chips/nrf52/src/nrf_radio.rs

```diff
@@ -544,19 +641,34 @@ impl Radio {
             tx_power: Cell::new(TxPower::ZerodBm),
             rx_client: OptionalCell::empty(),
             tx_client: OptionalCell::empty(),
+            tx_buf: TakeCell::empty(),
+            rx_buf: TakeCell::empty(),
+            addr: Cell::new(0),
+            addr_long: Cell::new([0x00; 8]),
+            pan: Cell::new(0),
+            cca_count: Cell::new(0),
+            cca_be: Cell::new(0),
+            random_nonce: Cell::new(0xDEADBEEF),
+            channel: Cell::new(radio::RadioChannel::DataChannel11),
+            transmitting: Cell::new(false),
         }
     }
 
-    fn tx(&self) {
-        let regs = &*self.registers;
-        regs.event_ready.write(Event::READY::CLEAR);
-        regs.task_txen.write(Task::ENABLE::SET);
-    }
-
     fn rx(&self) {
         let regs = &*self.registers;
         regs.event_ready.write(Event::READY::CLEAR);
+
+        if self.transmitting.get() {
+            self.tx_buf
+                .replace(self.set_dma_ptr(self.tx_buf.take().unwrap()));
```

Comment:
I'm not a fan of these `unwraps` at least change them to expect? 

Could be addressed in another PR

Reply:
I'm happy to do this, but I'm not exactly sure what needs to be done here. Is there an explainer of the difference between `unwrap`, `expect`, and `map`?

Best, 
DC

Reply:
Unwrap() will panic if the value in the cell is not there. Given Tock's reliability goals, we generally want to discourage code in the kernel which could lead to panics, as they bring the entire system down. Expect() will also panic if the value is not there, but allows you to pass a custom message, which makes it easier to at least trace where the error originated when debugging. 

Map() makes panic'ing impossible by requiring you to explicitly handle the case of the value not being present. So for the first unwrap in the above code example. you would do something like this (though the None case could vary depending on what should happen if the buffer is missing):
```
if self.transmitting.get() {
            match self.tx_buf.take() {
                    Some(buf) => {
                             self.tx_buf.replace(self.set_dma_ptr(buf));
                    },
                    None => debug!("Error: tx_buf missing"),
             }
// ...
```

Reply:
Ah I see. I thought this was more in response to the use of the direct setting of the DMA pointer. Fixed!